src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -2805,12 +2795,14 @@ DistObjectRelationId(void)
 								 true);
 	if (!OidIsValid(MetadataCache.distObjectRelationId))
 	{
+		Oid citusNamespaceId = get_namespace_oid("citus", false);
+
 		/*
 		 * We can only ever reach here while we are creating/altering our extension before
 		 * the table is moved to pg_catalog.
 		 */
 		CachedRelationNamespaceLookupExtended("pg_dist_object",
-											  CitusCatalogNamespaceId(),
+											  citusNamespaceId,
```

Comment:
Why are we replacing CitusCatalogNamespaceId() with a call made to get_namespace_oid()? The earlier is much cheaper.